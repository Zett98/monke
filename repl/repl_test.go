package repl

import (
	"bytes"
	"os"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func TestReplLoop(t *testing.T) {
	input := `
	let five = 5;
	let ten = 10;
	let add = fn(x, y) { x + y; }
	let result = add(five, ten);
	5 / 5 / -5;
	5 < 10 > 5;

	if (5 < 10) { return true; } else { return false; }

	10 == 10;
    10 != 9;
	let x 12 * 3
	`
	in := bytes.NewBufferString(input)
	out := bytes.NewBuffer(make([]byte, 0))
	t.Run("repl tests", func(t *testing.T) {
		Start(in, out)
		result := string(out.Bytes())
		snaps.MatchSnapshot(t, result)
	})
}
func TestMain(m *testing.M) {
	v := m.Run()

	// After all tests have run `go-snaps` will sort snapshots
	snaps.Clean(m, snaps.CleanOpts{Sort: true})

	os.Exit(v)
}
