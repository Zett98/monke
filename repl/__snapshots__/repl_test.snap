
[TestReplLoop/repl_tests - 1]
0
ERROR: type mismatch: BOOLEAN > INTEGER
true
true
true
Woops! We ran into some monkey business here!
 parser errors:
    expected next token to be `=`, got `INT` instead

---
